### PR TITLE
Eric/74 cicd functional testing

### PR DESCRIPTION
**Pull request by egieseke**
## Summary

Extend Expect tests to support running in an existing network to support automated network testing with Jenkins pipeline. 

## Test Plan

Tested with Jenkins CI/CD pipeline

### CODE REVIEW

File: test/e2e-go/cli/goal/cicd_expect/goalAlgonetStatusTest.exp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    puts "TEST_DATA_DIR: $TEST_DATA_DIR"
+
+    set TIME_STAMP [clock seconds]
+
+    set TEST_ROOT_DIR $TEST_ALGO_DIR/data
+
+    set NODE_NAME [exec ls $TEST_ALGO_DIR/data]
+    set TEST_PRIMARY_NODE_DIR $TEST_ROOT_DIR/$NODE_NAME
+
+    exec sleep 5
```
Comment (tsachiherman) [inline test/e2e-go/cli/goal/cicd_expect/goalAlgonetStatusTest.exp]: I'm not opposing to sleep for 5 seconds.. but why do we need to ?

File: test/e2e-go/cli/goal/cicd_expect/goalAlgonetStatusTest.exp [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    # use goal node status command to wait for round 3
+    ::AlgorandGoal::WaitForRound 5 $TEST_PRIMARY_NODE_DIR
```
Comment (tsachiherman) [inline test/e2e-go/cli/goal/cicd_expect/goalAlgonetStatusTest.exp]: Are we waiting for round 3 as in the comment or round 5 ?

File: test/e2e-go/cli/goal/cicd_expect/goalAlgonetStatusTest.exp [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+log_user 1
+
+if { [catch {
+
+    set ::GLOBAL_TEST_ALGO_DIR ""
+    set ::GLOBAL_TEST_ROOT_DIR ""
+    set ::GLOBAL_NETWORK_NAME ""
+
+    source  ../expect/goalExpectCommon.exp
+    set TEST_ALGO_DIR [lindex $argv 0]
```
Comment (tsachiherman) [inline test/e2e-go/cli/goal/cicd_expect/goalAlgonetStatusTest.exp]: Do we need to add testing for valid input argument values ?

File: test/e2e-go/cli/goal/cicd_expect/goalAlgonetStatusTest.exp (hunk 1 above)
Comment (egieseke) [reply test/e2e-go/cli/goal/cicd_expect/goalAlgonetStatusTest.exp]: This is required to allow the nodes to initialize.

File: test/e2e-go/cli/goal/cicd_expect/goalAlgonetStatusTest.exp (hunk 2 above)
Comment (egieseke) [reply test/e2e-go/cli/goal/cicd_expect/goalAlgonetStatusTest.exp]: Fixed comment to indicate round 5.

File: test/e2e-go/cli/goal/cicd_expect/goalAlgonetStatusTest.exp (hunk 3 above)
Comment (egieseke) [reply test/e2e-go/cli/goal/cicd_expect/goalAlgonetStatusTest.exp]: Added check to make sure that the provided directories exist.

**Comment by egieseke** [comment]
Made updates based on Tsachi's suggestions

**Pull request merged**